### PR TITLE
feat: add all variants of login page

### LINKED ISSUE #53: [Layout] Login Page
We would like to create a new section in sidebar called Layouts. First one should be Login Page. It should constist of a card with a login / register form.

Visual Concept:
https://tailwindui.com/components/application-ui/forms/sign-in-forms  - No. 2 Card design

Prerequisites
- Create ready to copy Card component (can be taken from tailwindUI if no designs available)

Details
- Create ready to copy Login Template View
- For now can look like the customized version of the tailwindUI one
- with changed logo and texts
- variant with login & password
- variant with oauth
- variant with only google oauth (like d.selleo now)

### PR DESCRIPTION
**Pull request by DawidHaratyk**
closes #53

### CODE REVIEW

**Review by k1eu — CHANGES_REQUESTED** [review]
https://github.com/Selleo/design-system/issues/53
- There's no card component created
- Please extract each icon from the views into own files as we already do with the other icons

**Comment by k1eu** [comment]
image
Google version has twitter icon

File: packages/selleo-design-core/src/LoginView.tsx [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+                >
+                  Forgot your password?
+                </a>
+              </div>
+            </div>
+
+            <Button
+              variant="primary"
+              size="normal"
+              class="w-[100%] text-center bg-indigo-600 mt-3 py-1 text-white"
```
Comment (k1eu) [inline packages/selleo-design-core/src/LoginView.tsx:83]: please don't override button colors

**Review by k1eu — CHANGES_REQUESTED** [review]

File: packages/selleo-design-core/src/LoginView.tsx (hunk 1 above)
Comment (DawidHaratyk) [reply packages/selleo-design-core/src/LoginView.tsx:83]: You're right, I'll change that

**Comment by PoslinskiNet** [comment]
@DawidHaratyk 

`Sign in to your account` -> `ApplicationName`
`Twitter icon` + "Text" -> `Google Icon + "Sign in"`

Also - why we are not using our buttons from the design system (orange one)?

**Comment by DawidHaratyk** [comment]
> 

I've implemented these changes now, about buttons from a design system, I've implemented these as well

**Comment by DawidHaratyk** [comment]
> @DawidHaratyk
> 
> `Sign in to your account` -> `ApplicationName` i.e. Dashboard `Twitter icon` + "Text" -> `Google Icon + "Sign in"` `Text about trial` -> `Short description of the application` i.e. Coworker Portal
> 
> Also - why we are not using our buttons from the design system (orange one)?

I've implemented these changes now, about buttons from a design system, I've implemented these as well

File: packages/selleo-design-core/src/Button.tsx [hunk 2]
```diff
@@ -1,54 +1,63 @@
...
 
-  const iconSize = size === "small" ? 16 : 24;
+  const iconSize = size === 'small' ? 16 : 24;
 
   return (
     <button class={buttonClasses} {...rest}>
       <div class={contentClasses}>
         {IconStart && <IconStart width={iconSize} height={iconSize} />}
-        Text
+        {children}
```
Comment (k1eu) [inline packages/selleo-design-core/src/Button.tsx:58]: it does break on Button page due to the missing content, can you try to find a way to fix it ? :D

File: packages/selleo-design-core/src/Button.tsx [hunk 3]
```diff
@@ -1,54 +1,63 @@
...
         {IconEnd && <IconEnd width={iconSize} height={iconSize} />}
-        {iconTest ? iconTest : ""}
+        {iconTest ? iconTest : ''}
```
Comment (k1eu) [inline packages/selleo-design-core/src/Button.tsx:60]: iconTest || null

**Review by k1eu — CHANGES_REQUESTED** [review]

File: packages/selleo-design-core/src/Button.tsx (hunk 2 above)
Comment (DawidHaratyk) [reply packages/selleo-design-core/src/Button.tsx:58]: Yes, if there is no text content provided, default text appear
{children || "Text"}
Have you that kind of solution on mind? :D

File: packages/selleo-design-core/src/Button.tsx (hunk 3 above)
Comment (DawidHaratyk) [reply packages/selleo-design-core/src/Button.tsx:60]: Right, it's better :)

File: packages/selleo-design-core/src/Button.tsx (hunk 2 above)
Comment (k1eu) [reply packages/selleo-design-core/src/Button.tsx]: if it works like that sure :D but we have seen issues with the `children` props so I'm asking :D

Comment (DawidHaratyk) [reply packages/selleo-design-core/src/Button.tsx]: Now in a button component "children" are rendering as expected :)

File: packages/selleo-design-core/src/LoginView.tsx [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+  GoogleIcon,
+  SelleoLogoIcon,
+  TwitterIcon,
+} from './icons';
+import { Input } from './Input';
+
+export function LoginWithPasswordView() {
+  return (
+    <Card>
+      <div className="flex min-h-full items-center justify-center py-12 px-4 sm:px-6 lg:px-8">
```
Comment (k1eu) [inline packages/selleo-design-core/src/LoginView.tsx:17]: Can that be merged, so Card Just takes those styles as classaName?

**Review by k1eu — CHANGES_REQUESTED** [review]

File: packages/selleo-design-core/src/LoginView.tsx (hunk 4 above)
Comment (DawidHaratyk) [reply packages/selleo-design-core/src/LoginView.tsx:17]: Yes, I'll merge it, I've just added possibility to add additional styles to the Card component and it takes className as a prop to add own styles :D

**Pull request merged**